cmd/ltkeycloak/from_mattermost.go

```diff
@@ -40,12 +40,43 @@ func migrateUser(worker *workerConfig, user *model.User) error {
 		// Ignore already existing users
 		if apiErr, ok := err.(*gocloak.APIError); ok && apiErr.Code == 409 {
 			mlog.Debug("user already exists in keycloak", mlog.String("username", user.Username))
-			return nil
+
+			// If the `--migrate-all-users` flag is set, we need to get the user ID of the already existing user from
+			// keycloak to set up the auth data of the mattermost user.
+			if worker.migrateAllUsers {
+				mlog.Debug("trying to fetch the user from keycloak to update the mattermost user", mlog.String("username", user.Username))
+				users, err := worker.keycloakClient.GetUsers(
+					ctx,
+					worker.keycloakToken.AccessToken,
+					worker.keycloakRealm,
+					gocloak.GetUsersParams{
+						Username: &user.Username,
+						Exact:    gocloak.BoolP(true),
+					},
+				)
+				if err != nil {
+					mlog.Error("failed to get user from keycloak", mlog.String("err", err.Error()))
+					worker.errorsChan <- err
+					return err
+				}
+
+				if len(users) != 1 {
+					err = fmt.Errorf("expected 1 user, got %d", len(users))
+					mlog.Error("got more users than expected in keycloak", mlog.String("username", user.Username), mlog.Int("users_count", len(users)))
```

Comment:
We typically avoid logging and returning an error. If we are returning the error, then let's avoid logging it here, and let it be logged at a higher layer.